reflection.md

```diff
@@ -15,4 +18,40 @@ Write a few brief paragraphs that discuss:
    you implement it _or_ what features of Scala would prevent you from
    implementing it? (You don't have to write code for this part. You could say
    "I would use literal extension to..." or "Scala's rules for valid
-   identifiers prevent...")
\ No newline at end of file
+   identifiers prevent...")
+
+
+The easiest things to say are things with repetition. For me, regular expressions can be strong when using union operators with the Kleene star. One can then specify patterns that can be arbitrarily long (with the Kleene star) and relatively complex using the union operator within the Kleene star. In fact, the family of operators `<*>`, `<+>`, and `{n}` are all related to doing things with repetition. As a result, we can be very specific about our repetitions.
+
+On the other hand, it seems that it is fairly difficult to write regular expressions when there are patterns that aren't easily represented using repetition. For example, the code we were given had the following example:
+
+```
+val digit = '0' || '1' || '2' || '3' || '4' || '5' || '6' || '7' || '8' || '9' 
+```
+
+This code made me a bit sad because there is an obvious pattern, but we were forced to union 10 literals. In general, union and concatenation operations can create clunky regular expressions. Not that they by themselves are bothersome, but when things get cumbersome, I believe these two operations are deeply involved.
+
+Another trait about our DSL that slightly bothers me is how we have to explicitly use parentheses to correctly express our regular expression:
+
+```
+val pattern = "42" || ( ('a' <*>) ~ ('b' <+>) ~ ('c'{3}))
+```
+
+In this code, we are forced to put three pairs of parentheses to write this regular expression. It would be interesting to see if we can create some sort of rules that allows us to remove parentheses. Some parentheses can be good for clarifying the regular expressions, but for those that seem redundant, removing parentheses could be nice.
```

Comment:
Maybe not having the parentheses would give the parser more work. For example, the following expression looks more cumbersome than it has to be, especially since even grade school students know that one multiplies before one adds:
(1*2)+(3*4) 

However, I think that allowing a removal of the parentheses would mean that you have to write a parser that understands how to interpret 1*2+3*4 as (1*2)+(3*4). I'm hypothesizing that there might be a trade-off between the complexity in writing RegularExpressions and the complexity in the parser that interprets the RegularExpression?